code_search/demo/cs-demo-1103/ks_app/components/params.libsonnet

```diff
@@ -0,0 +1,109 @@
+{
+  global: {},
+  components: {
+    // Component-level parameters, defined initially from 'ks prototype use ...'
+    // Each object below should correspond to a component in the components/ directory
+    "pytorch-operator": {
+      cloud: 'null',
+      deploymentNamespace: 'null',
+      deploymentScope: 'cluster',
+      disks: 'null',
+      name: 'pytorch-operator',
+      namespace: 'null',
+      pytorchDefaultImage: 'null',
+      pytorchJobImage: 'gcr.io/kubeflow-images-public/pytorch-operator:v0.3.0',
+      pytorchJobVersion: 'v1alpha2',
+    },
+    ambassador: {
+      ambassadorImage: 'quay.io/datawire/ambassador:0.37.0',
+      ambassadorServiceType: 'ClusterIP',
+      cloud: 'gke',
+      name: 'ambassador',
+      statsdImage: 'quay.io/datawire/statsd:0.37.0',
+      statsdSinkImage: 'prom/statsd-exporter:v0.6.0',
+    },
+    jupyterhub: {
```

Comment:
So you're including JupyterHub, is this deployment meant for research/experiments as well as serving the demo?

If that's the case I would advocate for NFS to be included as well. Perhaps there's an alternative solution using PV's but NFS has been invaluable for me thus far being the place where all of my wip code and notebook are stored - if the notebook container resets, which happens often at least for me, then nothing is lost.

Perhaps gcsfusemounting a bucket is as good or better (in which case the nb container would need gcsfuse). In support of this latter approach I did see one or two times my NFS go down or accidentally get deleted when I re-deployed and lost some stuff so GCS is probably safer but you might also view that as a bug to fix and user error to avoid, respectively.

Reply:
We can add an NFS mount later if needed. 
https://www.kubeflow.org/docs/guides/gke/cloud-filestore/

Right now we don't have an immediate need so I didn't bother to set it up yet.